proxy/certs/certs.go

```diff
@@ -215,9 +217,35 @@ func (s *RemoteCertSource) Remote(instance string) (cert *x509.Certificate, addr
 		logging.Errorf("%v", err)
 		logging.Errorf("WARNING: specifying the correct region in an instance string will become required in a future version!")
 	}
-	if len(data.IpAddresses) == 0 || data.IpAddresses[0].IpAddress == "" {
+
+	if len(data.IpAddresses) == 0 {
 		return nil, "", "", fmt.Errorf("no IP address found for %v", instance)
 	}
+
+	// Add the option to choose which IP address to use based on user input ip_address_type
+	// Assume IP address type order is not fixed, we need to find the corresponding IP address by iterating IPAddresses
+	flagTypeName := "ip_address_type"
```

Comment:
Instead of needing to mess with the flags like this, I think it would be better to have a field in the RemoteCertSource struct that specified the ip address type preference.

It could then be set by the flag in sidecar mode, as well as settable via code in the library mode.